src/simulator_core/entities/assets/production_cluster.py

```diff
@@ -221,38 +219,34 @@ def simulation_performed(self) -> bool:
             False otherwise.
         """
         if self.pandapipes_net.res_circ_pump_mass is AttributeError:
-            # TODO: Implement specific error
+            #  TODO: Implement specific error
             return False
         else:
-            # Retrieve the setpoints
+            #  Retrieve the setpoints
             return True
 
-    def get_setpoints(self) -> Dict:
+    def get_setpoints(self) -> Dict[str, float]:
         """Get the setpoints of the asset.
 
         :return Dict setpoints: The setpoints of the asset in a dictionary,
             as "property_name": value pairs.
         """
         # Return the setpoints
-        if self.simulation_performed():
```

Comment:
Where do you now perform the check whether the simulation is performed? If you remove this statement and you have initialized the network but call the results you'll get an error from pandapipes and not from our side?

Reply:
Adjustted it